src/hotspot/share/gc/parallel/psOldGen.hpp

```diff
@@ -135,8 +135,12 @@ class PSOldGen : public CHeapObj<mtGC> {
   void resize(size_t desired_free_space);
 
   HeapWord* allocate(size_t word_size) {
-    HeapWord* res = cas_allocate_noexpand(word_size);
-    return (res == NULL) ? expand_and_cas_allocate(word_size) : res;
+    HeapWord* res;
+    do {
+      res = cas_allocate_noexpand(word_size);
+      // Retry failed allocation if expand succeeds.
```

Comment:
"... but allocation did not." would be nice to be added to this comment to be complete.

Reply:
That's a "failed allocation".

Reply:
Agreed :)